### DISCUSSION TITLE
Request advice and insights on subtitles playback technology used

### DISCUSSION THREAD

**Discussion by JohnPaulHarold** [bbc/bigscreen-player#318] [Q&A] [answered]
Hello. I'm investigating options for the rendering of subtitles for DASH streams on non-MSE/EME based devices. I'm working in the UK. I was interested in this advice, https://bbc.github.io/bigscreen-player/api/tutorial-Subtitles.html#why-not-include-subtitles-in-the-manifest%3F

Do you have any additional documentation you could share about the approach you have taken, specifically regarding the code implementation. Or alternatively, could you point to the parts of the code critical to achieving this feature? You mention the sentinels, and I can see them in the code but do you also have any documents detailing them further?

My motivation is to offer subtitles for all possible streams and so far, device compatibility and or implementations is proving to be patchy.

I hope this is the correct place for such a question, and thank you for reading this.

**Comment by matt-stephenson** [comment, marked as answer]
Hello, thanks for your query about subtitles.

At the BBC, for our smart TV player, we always ‘side load’ subtitles, rather than include them as part of the mpd, mainly because we do not know how native players (e.g. non-MSE devices where we just pass an mpd to the media element) handle them. We would also not have control over the visual aspects of presentation. With the large number of devices we support, we value the consistency a side loading approach gives us.

We achieve this as detailed in the bigscreen-player documentation that you link to. For example, for VOD, we load the subs from a single XML file, conforming to the spec shown [here](https://www.bbc.co.uk/accessibility/forproducts/guides/subtitles#Example-EBU-TT-D-documentand), by giving the url of this XML to bigscreen-player on initialisation. The subtitles implementation then simply uses an interval to obtain the media element’s `currentTime` and renders corresponding subs (usually using imsc.js). Again, this solution is for consistency across the device estate (support for textTracks isn’t always there on some older devices). The solution is mainly found in https://github.com/bbc/bigscreen-player/tree/master/src/subtitles.

I’m unsure as to your question regarding sentinels - these are used in our non-MSE player, but sentinels themselves are for handling video state events, and not so relevant to subtitle presentation.

Hopefully this helps with some info on the way we are doing things, but as it’s always good to experiment when it comes to Smart TV development, there are some example streams provided by BBC R&D which do have embedded subtitle tracks in the mpds, if they would be helpful to see. This page does also includes the single XML version - https://rdmedia.bbc.co.uk/elephants_dream/

Thanks,
Matt

**Comment by JohnPaulHarold** [comment]
thank you for your reply. I must have got muddled about Sentinels, my apologies for the confusion.

Yes, the motivation for this question is solely about native DASH playback on TVs and STBs. My interest at the moment is mostly towards offering subtitles for live streams.

I will do some more experiments but probably mark your reply as the answer anon.
